salt/modules/salty_swarm.py

```diff
@@ -0,0 +1,269 @@
+# -*- coding: utf-8 -*-
+"""
+Docker Swarm Module using Docker's Python SDK
+=============================================
+
+:codeauthor: Tyler Jones <jonestyler806@gmail.com>
+
+.. versionadded:: 2017.7.2
+
+The Docker Swarm Module is used to manage and create Docker Swarms.
+
+Dependencies
+============
+
+- Docker installed on the host
+- Docker python sdk >= 2.5.1
+
+Docker Python SDK
+=================
+pip install -U docker
+
+More information: https://docker-py.readthedocs.io/en/stable/
+"""
+from __future__ import absolute_import
+import docker
+import salt.config
+import salt.loader
+import json
+__opts__ = salt.config.minion_config('/etc/salt/minion')
+__grains__ = salt.loader.grains(__opts__)
+client = docker.from_env()
+server_name = __grains__['id']
+
+
+def swarm_tokens():
+    '''
+    Get the Docker Swarm Manager or Worker join tokens
+    '''
+    client = docker.APIClient(base_url='unix://var/run/docker.sock')
+    service = client.inspect_swarm()
+    return service['JoinTokens']
+
+
+def swarm_init(advertise_addr=str,
+               listen_addr=int,
+               force_new_cluster=bool):
+    '''
+    Initalize Docker on Minion as a Swarm Manager
+    '''
```

Comment:
Each of these functions will need to have the arguments explained in the docstring and an commandline example is also required.

Reply:
I'll get this updated.